tests/test_tools.py

```diff
@@ -728,3 +729,35 @@ def test_group_member_ids(input, expected):
     actual = group_member_ids(input, "bg")
     assert actual == expected
 
+def test_fragment_retrieval():
```

Comment:
it is better to use the ``parameterize`` fixture than do it this way.  Please can you reach out to Zach or someone who knows how to use that? It was explained in the last group meeting.